### PR TITLE
Modify abstract classes that are static utilities (or currently function...

### PR DESCRIPTION
**Pull request by osmandin**
... as such)

https://jira.duraspace.org/browse/FCREPO-1323

http://sonar.fcrepo.org/drilldown/issues/1?&rule=pmd%3AAbstractClassWithoutAbstractMethod&rule_sev=MAJOR&severity=MAJOR#

### CODE REVIEW

**Comment by ajs6f** [comment]
Why are some of these guys made `final` and others not?

**Comment by osmandin** [comment]
Tthe final ones had only static methods. I wasn't sure if extension was intended for the other ones.

**Comment by ajs6f** [comment]
Makes sense. `AbstractService` and `AbstractResource` could be (and are) usefully extended, but `ServiceHelpers` has a private constructor-- I don't think it is meant for extension.

**Comment by ajs6f** [comment]
In Java8, we might want to make `AbstractService` and `AbstractResource` into interfaces with default methods.

File: fcrepo-http-commons/src/main/java/org/fcrepo/http/commons/AbstractResource.java [hunk 1]
```diff
@@ -37,12 +37,12 @@
...
 
 /**
- * Abstract superclass for Fedora JAX-RS Resources, providing convenience fields
+ * Superclass for Fedora JAX-RS Resources, providing convenience fields
  * and methods.
  *
  * @author ajs6f
  */
-public abstract class AbstractResource {
+public class AbstractResource {
```
Comment (awoods) [inline fcrepo-http-commons/src/main/java/org/fcrepo/http/commons/AbstractResource.java]: This class should have a private constructor.

File: fcrepo-kernel-impl/src/main/java/org/fcrepo/kernel/impl/services/AbstractService.java [hunk 2]
```diff
@@ -34,7 +34,7 @@
  * @author ajs6f
  * @since Feb 20, 2014
  */
-public abstract class AbstractService {
+public class AbstractService {
```
Comment (awoods) [inline fcrepo-kernel-impl/src/main/java/org/fcrepo/kernel/impl/services/AbstractService.java]: This class should have a private constructor.

File: fcrepo-kernel-impl/src/main/java/org/fcrepo/kernel/impl/services/ServiceHelpers.java [hunk 3]
```diff
@@ -189,4 +189,7 @@ public static long getRepositorySize(final Repository repository)
             session.logout();
         }
     }
+
```
Comment (awoods) [inline fcrepo-kernel-impl/src/main/java/org/fcrepo/kernel/impl/services/ServiceHelpers.java]: Constructor should be at the top of class, just below the instance variables.

File: fcrepo-kernel/src/main/java/org/fcrepo/kernel/services/functions/JcrPropertyFunctions.java [hunk 4]
```diff
@@ -139,4 +139,7 @@ public boolean apply(final Node node) {
             }
         }
     };
+
```
Comment (awoods) [inline fcrepo-kernel/src/main/java/org/fcrepo/kernel/services/functions/JcrPropertyFunctions.java]: Constructor should be at the top of class, just below the instance variables.

File: fcrepo-kernel/src/main/java/org/fcrepo/kernel/utils/ContentDigest.java [hunk 5]
```diff
@@ -98,4 +99,7 @@ private static String asString(final byte[] data) {
     public static URI missingChecksum() {
         return asURI("SHA-1", "missing");
     }
+
+    private ContentDigest() {
```
Comment (awoods) [inline fcrepo-kernel/src/main/java/org/fcrepo/kernel/utils/ContentDigest.java]: Constructor should be at the top of class, just below the instance variables.

File: fcrepo-kernel/src/main/java/org/fcrepo/kernel/utils/NamespaceTools.java [hunk 6]
```diff
@@ -105,4 +105,7 @@ public static void validatePath(final Session session, final String path) {
             }
         }
     }
+
```
Comment (awoods) [inline fcrepo-kernel/src/main/java/org/fcrepo/kernel/utils/NamespaceTools.java]: Constructor should be at the top of class, just below the instance variables.

File: fcrepo-http-commons/src/main/java/org/fcrepo/http/commons/AbstractResource.java (hunk 1 above)
Comment (ajs6f) [inline fcrepo-http-commons/src/main/java/org/fcrepo/http/commons/AbstractResource.java]: No, this class is extensible.

File: fcrepo-kernel-impl/src/main/java/org/fcrepo/kernel/impl/services/AbstractService.java (hunk 2 above)
Comment (ajs6f) [inline fcrepo-kernel-impl/src/main/java/org/fcrepo/kernel/impl/services/AbstractService.java]: No, this class is extensible.

Comment (awoods) [inline fcrepo-kernel-impl/src/main/java/org/fcrepo/kernel/impl/services/AbstractService.java]: ok

File: fcrepo-http-commons/src/main/java/org/fcrepo/http/commons/AbstractResource.java (hunk 1 above)
Comment (awoods) [inline fcrepo-http-commons/src/main/java/org/fcrepo/http/commons/AbstractResource.java]: ok

**Comment by awoods** [comment]
Resolved with: https://github.com/fcrepo4/fcrepo4/commit/1317ed16f91af3b9bbcbc28c3f9b4dfa80eb75bc

**Pull request closed without merging**